src/scf-release/src/acceptance-tests-brain/test-scripts/016_syslog_forwarding_test.rb

```diff
@@ -0,0 +1,124 @@
+#!/usr/bin/env ruby
+
+require_relative 'testutils'
+require 'json'
+require 'securerandom'
+require 'timeout'
+
+def show_env
+    ENV.sort.select { |k, v| k.start_with? 'SCF_LOG_' }.each do |k, v|
+        puts "#{k}=#{v}"
+    end
+end
+
+# We have some waits in this test, and want to clean things up properly
+# (especially the service broker) when we abort.  So we wrap a timeout so that
+# we get a minute to do any cleanup we need.
+Timeout::timeout(ENV.fetch('TESTBRAIN_TIMEOUT', '600').to_i - 60) do
+    SCF_LOG_HOST = ENV.fetch('SCF_LOG_HOST', '')
+    SCF_LOG_PORT = ENV.fetch('SCF_LOG_PORT', '514')
+    SCF_LOG_PROTOCOL = ENV.fetch('SCF_LOG_PROTOCOL', 'tcp')
+    if SCF_LOG_HOST.empty?
+        message = "SCF_LOG_HOST not set"
+        STDERR.puts "\e[0;1;31m#{message}\e[0m"
+        show_env
+        fail message
+    end
+
+    $KUBERNETES_NAMESPACE = ENV['KUBERNETES_NAMESPACE']
+
+    KUBERNETES_DOMAIN_SUFFIX = ".#{$KUBERNETES_NAMESPACE}.svc.#{ENV['KUBERNETES_CLUSTER_DOMAIN']}"
+    unless SCF_LOG_HOST.end_with? KUBERNETES_DOMAIN_SUFFIX
+        message = "SCF_LOG_HOST (#{SCF_LOG_HOST}) does not end with cluster domain (#{KUBERNETES_DOMAIN_SUFFIX})"
+        STDERR.puts "\e[0;1;31m#{message}\e[0m"
+        show_env
+        fail message
+    end
+
+    $LOG_SERVICE_NAME = SCF_LOG_HOST[0...-KUBERNETES_DOMAIN_SUFFIX.length]
+
+    # Report progress to the user; use as printf
+    def status(fmt, *args)
+        printf "\n\e[0;32m#{fmt}\e[0m\n", *args
+    end
+
+    # Report problem to the user; use as printf
+    def trouble(fmt, *args)
+        printf "\n\e[0;31m#{fmt}\e[0m\n", *args
+    end
+
+    $RUN_SUFFIX = SecureRandom.hex(16)
```

Comment:
I might reduce this to 8. The hex doubles the output, making it 32 currently.

The last jenkins reported

```
15:46:43 + kubectl run 11280387a7c84f769917f9c5b1214b90 --namespace pr-2027-5-scf --command --port 514 --expose --image=splatform/scf-acceptance-tests-brain:4dae05e3949d150f2acddeff10caf8a5edbaf226 --labels=brains=11280387a7c84f769917f9c5b1214b90.1f4e935891d083b1bbf46adb7cfaead4 -- /bin/sh -c zypper --non-interactive install /usr/bin/socat /usr/bin/logger && /usr/bin/socat TCP-LISTEN:514,fork 'EXEC:/usr/bin/logger --socket-errors=off --stderr --tag ""'
15:46:43 The Deployment "11280387a7c84f769917f9c5b1214b90" is invalid: 
15:46:43 * metadata.labels: Invalid value: "11280387a7c84f769917f9c5b1214b90.1f4e935891d083b1bbf46adb7cfaead4": must be no more than 63 characters
15:46:43 * spec.selector.matchLabels: Invalid value: "11280387a7c84f769917f9c5b1214b90.1f4e935891d083b1bbf46adb7cfaead4": must be no more than 63 characters
15:46:43 * spec.selector: Invalid value: v1.LabelSelector{MatchLabels:map[string]string{"brains":"11280387a7c84f769917f9c5b1214b90.1f4e935891d083b1bbf46adb7cfaead4"}, MatchExpressions:[]v1.LabelSelectorRequirement(nil)}: invalid label selector.
```

I get a failing kube command under vagrant as well somewhere, but later I think. It still has this long hex value in names, and I worry that this is the problem.

If not I will let jenkins run on the reduced form to see if that then matches what I see.

Reply:
Yep. uuidgen yields 32 characters, plus 32 char suffix, plus dots => Too long.
My reduction of the suffix made that pass.
Still a problem, uuidgen output may start with a digit. The service name must start with an alphabetical char.

Pushed a fix, also reduced the random part to 16 chars too, to keep under the limits.